File: engine/engine-model/src/main/scala/server/movement/ChairMovement.scala
package server.movement

import java.awt.Point

import tools.data.output.LittleEndianWriter

class ChairMovement(private var _type: Int, private var _position: Point, private var _duration: Int, private var _newState: Int, private var _fh: Int) extends AbstractLifeMovement(_type = _type, _position = _position, _duration = _duration, _newState = _newState) {
  def fh: Int = _fh

  override def serialize(lew: LittleEndianWriter): Unit = {
    lew.write(theType)
    lew.writeShort(position.x)
    lew.writeShort(position.y)
    lew.writeShort(fh)
    lew.write(newState)
    lew.writeShort(duration)
  }
}

File: engine/engine-model/src/main/java/server/movement/ChairMovement.java
package server.movement;

import java.awt.Point;

import tools.data.output.LittleEndianWriter;

public class ChairMovement extends AbstractLifeMovement {
   private final int fh;

   public ChairMovement(int theType, Point position, int duration, int newState, int fh) {
      super(theType, position, duration, newState);
      this.fh = fh;
   }

   public int fh() {
      return fh;
   }

   @Override
   public void serialize(LittleEndianWriter lew) {
      lew.write(theType());
      lew.writeShort(position().x);
      lew.writeShort(position().y);
      lew.writeShort(fh);
      lew.write(newState());
      lew.writeShort(duration());
   }
}
